project/make/gccgo

```diff
@@ -0,0 +1,35 @@
+#!/bin/bash
+set -e
+
+DEST=$1
+BINARY_NAME="docker-$VERSION"
+BINARY_EXTENSION="$(binary_extension)"
+BINARY_FULLNAME="$BINARY_NAME$BINARY_EXTENSION"
+
+# Cygdrive paths don't play well with go build -o.
+if [[ "$(uname -s)" == CYGWIN* ]]; then
+	DEST=$(cygpath -mw $DEST)
+fi
+
+source "$(dirname "$BASH_SOURCE")/.go-autogen"
+
+if [ "$STATIC_DOCKER" == "true" ]; then
+	EXTLDFLAGS_STATIC_DOCKER="-Wl,--no-export-dynamic $EXTLDFLAGS_STATIC_DOCKER"
+else
+	EXTLDFLAGS_STATIC_DOCKER="-static-libgo $(echo $EXTLDFLAGS_STATIC_DOCKER | sed "s/$EXTLDFLAGS_STATIC//")"
```

Comment:
Can a non-static Docker daemon compiled via GCCGO actually work?  It seems like we're doing extra work here to get a completely static binary versus a mostly-static binary, and I'm confused why we're even bothering -- that seems out of scope.  See `hack/make/dynbinary` vs `hack/make/binary` for how we handle this without GCCGO (ie, a separate smaller `dockerinit` binary that's completely static, and then the `docker` binary can be completely dynamic).